pyproject.toml

```diff
@@ -84,10 +84,10 @@ all = [
     "bottleneck",
     "ipython>=4.2",
     "pytest>=7.0",
-    "typing_extensions>=3.10.0.1",
     "fsspec[http]>=2023.4.0",
     "s3fs>=2023.4.0",
     "pre-commit",
+    "typing_extensions>=4.0.0 ; python_version < '3.11'",
```

Comment:
It is good to try to keep track of why `typing_extensions` is needed, but it is also important to record the reason in commit messages.